libr/bin/p/bin_fs.c

```diff
@@ -53,18 +53,15 @@ static bool check_bytes(const ut8 *buf, ut64 length) {
 	return p != NULL;
 }
 
-static void * load_bytes(RBinFile *bf, const ut8 *buf, ut64 sz, ut64 loadaddr, Sdb *sdb){
-	if (check_bytes (buf, sz)) {
-		return R_NOTNULL;
-	}
-	return NULL;
+static bool load_bytes(RBinFile *bf, void **bin_obj, const ut8 *buf, ut64 sz, ut64 loadaddr, Sdb *sdb){
+	return check_bytes (buf, sz);
 }
 
 static bool load(RBinFile *bf) {
 	const ut8 *bytes = bf ? r_buf_buffer (bf->buf) : NULL;
 	ut64 sz = bf ? r_buf_size (bf->buf): 0;
 	ut64 la = (bf && bf->o) ? bf->o->loadaddr: 0;
-	return load_bytes (bf, bytes, sz, la, bf? bf->sdb: NULL) != NULL;
+	return load_bytes (bf, &bf->o->bin_obj, bytes, sz, la, bf? bf->sdb: NULL);
```

Comment:
?

Reply:
Problem is that `load_bytes()` is used through a function pointer declared in `r_bin.h`. Thus, I _think_ I can't use different function signatures on their definitions on `bin_*.c`, even though they are static, because it seems to offer some kind of polymorphism to the binary loading logic.

Reply:
I'm saying that you shouldn't pass bf->o->bin_obj (and you should check for nulls in load_bytes before assigning it). In the original code, bf->o->bin_obj was not set, in your code it *may* be  (if check_bytes will set it in the future)

Reply:
As `load_bytes()` doesn't change `bf->o->bin_obj` and it probably will once it grows to do more than call `check_bytes()`, isn't it reasonable leaving it there in order to ease further extension?

Reply:
I understood about null checking before setting the value of `bf->o->bin_obj`. But I didn't understand about not passing it `load_bytes()`, as we have to keep the function signature.

Reply:
I still didn't change those functions to check `bf->o->bin_obj` for null because I thought this PR is already too big, and I was afraid of changing too much at once. I can do it in a future PR or in this one, if you deem appropriate.

Reply:
ok no problem

Reply:
No problem if I defer that or no problem if I change more? Should I wait for this to be merged and work on those enhancements on a next PR or keep on working on it?

Reply:
Ok, then I'll prepare another PR to further refactor those things after the merge of this one.